docs/CHAT_ADAPTER.md

```diff
@@ -0,0 +1,110 @@
+
+### Updating activities
+
+Web Chat uses 2 different IDs to identify if 2 activities are the same or not:
+
+-  `activity.channelData.clientActivityID` is an ID set by Web Chat on all outgoing activity from the current session, ignored if the value is falsy
+-  `activity.id` is an ID set by the chat platform
+
+If either one of the IDs are the same, Web Chat will consider both are identifying the same activity. And it will replace the activity with last writer win strategy.
+
+### Sorting activities
+
+`activity.timestamp` is a service-assigned timestamp and is used to reorder activities in the transcript.
+
+For recent outgoing activities which service-assigned timestamp is not received from the server yet, Web Chat will use a timestamp with an estimated server clock.
+
+If an activity is updated with a different `activity.timestamp` value, Web Chat will honor the updated timestamp by moving the activity to a different location.
+
+### Delaying activities
+
+For accessibility reasons, activities must be placed on the screen based on their chronological order. As screen reader pick up activity at the time it was placed on the screen, it is not possible to reorder and sort activities once they are on-screen.
+
+However, the chat service send activities over the network and chat adapter may receive activities with a slightly out-of-order fashion.
+
+Web Chat use `activity.replyToId` as a hint for the order of messages. Please read more about it [here](https://github.com/microsoft/BotFramework-WebChat/blob/main/docs/ACCESSIBILITY.md#ux-message-ordering).
+
+If the chat service is not able to provide the `replyToId` field, please make sure activities sent through the `activity$` Observable is in chronological order.
+
+### Conversation history
+
+Currently, Web Chat do not differentiate whether an activity is from previous or current conversation.
+
+## `connectionStatus$: Observable<ConnectionStatus>`
+
+An Observable to notify the state of the connection:
+
+-  `0` means uninitialized
+-  `1` means connecting
+-  `2` means connected and online
+-  `3` means the token has expired
+-  `4` means the initial connection failed to establish, or connection is interrupted
+-  `5` means the connection has ended
+
+When connection is interrupted, the chat adapter should notify Web Chat by sending `4`. It is up to the chat adapter to reconnect. When the connection is reestablished, it should notify Web Chat by sending `2`.
+
+## `postActivity(activity: Activity): Observable<string>`
+
+A function callback for posting an outgoing activity to the chat service. Web Chat will call this function and expect the service to return a service-assigned activity ID.
+
+Although this function only returns a single ID when the activity is successfully sent, historically, the function return Observable instead of Promise.
```

Comment:
```suggestion
Although this function only returns a single ID when the activity is successfully sent, historically, the function returns an Observable instead of a Promise.
```